Test/PacketType/IcmpV6PacketTest.cs

```diff
@@ -69,22 +68,131 @@ public void IcmpV6Parsing()
+
+            Assert.True(neighborAdvertisementPacket.Router);
+            Assert.True(neighborAdvertisementPacket.Override);
+            Assert.False(neighborAdvertisementPacket.Solicited);
+
+            neighborAdvertisementPacket.Router = false;
+            Assert.False(neighborAdvertisementPacket.Router);
+            Assert.True(neighborAdvertisementPacket.Override);
+            Assert.False(neighborAdvertisementPacket.Solicited);
+
+            Assert.AreEqual(1, neighborAdvertisementPacket.OptionsCollection.Count);
+            var option = neighborAdvertisementPacket.OptionsCollection[0];
+            Assert.IsInstanceOf(typeof(NdpLinkLayerAddressOption), option);
+
+            var ndLinkLayerAddressOption = (NdpLinkLayerAddressOption) option;
+            Assert.AreEqual(PhysicalAddress.Parse("C20054F50000"), ndLinkLayerAddressOption.LinkLayerAddress);
+
+            // Router advertisement.
+            dev.GetNextPacket(out PacketCapture ra);
+            var raRawCapture = ra.GetPacket();
+
+            var pRa = Packet.ParsePacket(raRawCapture.GetLinkLayers(), raRawCapture.Data);
+            Assert.IsNotNull(pRa);
+
+            var routerAdvertisementPacket = pRa.Extract<NdpRouterAdvertisementPacket>();
+            Assert.AreEqual(0, routerAdvertisementPacket.ReachableTime);
+            routerAdvertisementPacket.ReachableTime = 1234;
+            Assert.AreEqual(1234, routerAdvertisementPacket.ReachableTime);
+
+            Assert.AreEqual(1800, routerAdvertisementPacket.RouterLifetime);
+            Assert.AreEqual(64, routerAdvertisementPacket.CurrentHopLimit);
+            Assert.False(routerAdvertisementPacket.ManagedAddressConfiguration);
+            Assert.False(routerAdvertisementPacket.OtherConfiguration);
+
+            Assert.AreEqual(3, routerAdvertisementPacket.OptionsCollection.Count);
+
+            var prefixOption = routerAdvertisementPacket.OptionsCollection.Find(x => x is NdpLinkPrefixInformationOption) as NdpLinkPrefixInformationOption;
+            Assert.NotNull(prefixOption);
+
+            Assert.AreEqual(64, prefixOption.PrefixLength);
+            Assert.True(prefixOption.OnLink);
+            Assert.True(prefixOption.AutonomousAddressConfiguration);
+            Assert.AreEqual(IPAddress.Parse("2001:db8:0:1::"), prefixOption.Prefix);
+
+            var mtuOption = routerAdvertisementPacket.OptionsCollection.Find(x => x is NdpMtuOption) as NdpMtuOption;
+            Assert.NotNull(mtuOption);
+
+            Assert.AreEqual(1500, mtuOption.Mtu);
+            mtuOption.Mtu = 1400;
+            Assert.AreEqual(1400, mtuOption.Mtu);
+
+            dev.Close();
+
+            // Redirect message.
+            dev = new CaptureFileReaderDevice(NUnitSetupClass.CaptureDirectory + "ipv6_redirect.pcap");
+            dev.Open();
+
+            for (int i = 0; i < 10; i++)
+            {
+                dev.GetNextPacket(out _);
+            }
```

Comment:
@kayoub5 Is there a better way to immediately go to X packet?

Reply:
Not that I can think of

Reply:
Would be a useful method to add perhaps, or an additional parameter to skip X packets.

Reply:
no performance benefit(the only way to skip a packet is to actually read it) , and could be done as extention method

Reply:
Just for convenience, I assume there are more cases where it would be useful to be able to skip some packets 😊